blocks/image-link-slider/image-link-slider.js

```diff
@@ -0,0 +1,106 @@
+// Map of API names and their respective endpoint URLs
+import { buildBlock, decorateBlock, loadBlock } from '../../scripts/aem.js';
+import { handleSocialShareClick } from '../../scripts/scripts.js';
+import { callAPI } from '../../scripts/mockapi.js';
+
+function renderImageLinkVariant({ data }, carouselItems) {
+  data.forEach((item) => {
+    const slide = document.createElement('li');
+    slide.classList.add('carousel-slide');
+    slide.innerHTML = `
+                    <div class="carousel-slide-image">
+                        <img src="${item.finImage}" alt="${item.finTitle}">
+                    </div>
+                    <div class="carousel-slide-content">
+                        <h3><a href="${item.finLink}">${item.finTitle}</a></h3>
+                        <div class="carousel-slide-content-footer copyright">
+                        <div>
+                            <div>${item.finPoweredBy}</div>
+                            <div>${item.finPoweredOn}</div>
+                        </div>
+                        <div class="socialshare">
+                            <a class="social-share">
+                                <img src="/icons/gray-share-icon.svg" alt="Social Share" >
+                            </a>
+                        </div>
+                    </div>
+                    </div>
+                `;
+    carouselItems.append(slide);
+  });
+}
+
+async function loadCarousel(block, carouselItems) {
```

Comment:
I think we should move this code to bock-utils.

Reply:
I feel it is very specific to Image-Link-Slider (or carousel). Do you see it is generic enough to fit in your case too?